shared/tinyusb/mp_usbd.c

```diff
@@ -49,12 +49,17 @@ void mp_usbd_task_callback(mp_sched_node_t *node) {
 
 extern void __real_dcd_event_handler(dcd_event_t const *event, bool in_isr);
 
+// Provide stub for ports that don't have/need the wake main task function.
+void __attribute((weak)) mp_hal_wake_main_task_from_isr(void) {
```

Comment:
I suggest to do this slightly differently and force all ports using these shared TinyUSB bindings to define this function.  The ports can just add the following to their `mphalport.h`:
```c
static inline void mp_hal_wake_main_task_from_isr(void) {
    // Nothing needs to be done here.
}
```

Reasons:
1. Reduce code size and don't need to call a useless empty function (see rp2 code size report).
2. Don't rely on weak linking, which may not be that portable and could lead to issues if the wrong function is linked instead.
3. Make it explicit that a port doesn't need to implement anything for this MP HAL function, and provide a clear place to add it if needed (self documenting).

Reply:
Thanks @dpgeorge I'm not a huge fan on boilerplate but yes it's true weak functions can be a little unclear / non-deterministic, it's hard to be certain which version is actually used. 

I considered using a define instead to set if this function should be needed, but that's hard to document / describe as well. 

This probably is the best compromise, I've implemented it for each of the TinyUSB ports now.

Reply:
> I considered using a define instead to set if this function should be needed, but that's hard to document / describe as well.

Yeah, I had the same thoughts as you.

Thanks for updating!